core/src/main/java/io/seata/core/rpc/netty/ChannelManager.java

```diff
@@ -399,7 +400,7 @@ public static Channel getChannel(String resourceId, String clientId, boolean try
             }
         }
 
-        if (resultChannel == null && tryOtherApp) {
+        if (resultChannel == null && BranchType.TCC == branchType) {
```

Comment:
为什么要改变入参，而不是在由调用方决定是否开启tryOtherApp
Why change the entry parameter when it's up to the caller to turn on tryOtherApp?

Reply:
i feel the place convention over configuration is better, method signature is better understood without loss of flexibility.
这个地方感觉用 预定大于配置 的方法更好一点，不失灵活性的同时方法签名也更好理解.

Reply:
What about tryOtherApp in AT transaction mode? The original logic is that only AT mode can go here, now only TCC mode can go here?

Reply:
> > What about tryOtherApp in AT transaction mode? The original logic is that only AT mode can go here, now only TCC mode can go here?
> 
> yeah, because of TCC rpc mode regiter branch in constomer, send order of TC two stage to provider. so it can't find the channel as normal, so tcc model will try other app. @funky-eyes 's opinion there may be a risk in at to find a same resource diff app channel, ex: two instance have status.
> 
> 是的，因为TCC RPC模式是在消费者端注册分支，TC在二阶段下发指令给服务者端。导致通过消费者端的app去找服务者端的channel的时候会失败，所以tcc 模式会尝试其他app channel. @funky-eyes 的建议在AT模式下发指令到相同资源不同app的端可能存在风险, 比如两台实例可能有状态的情况下.

Maybe this is not the scope of this bugfix.

Reply:
> > > What about tryOtherApp in AT transaction mode? The original logic is that only AT mode can go here, now only TCC mode can go here?
> > 
> > 
> > yeah, because of TCC rpc mode regiter branch in constomer, send order of TC two stage to provider. so it can't find the channel as normal, so tcc model will try other app. @funky-eyes 's opinion there may be a risk in at to find a same resource diff app channel, ex: two instance have status.
> > 是的，因为TCC RPC模式是在消费者端注册分支，TC在二阶段下发指令给服务者端。导致通过消费者端的app去找服务者端的channel的时候会失败，所以tcc 模式会尝试其他app channel. @funky-eyes 的建议在AT模式下发指令到相同资源不同app的端可能存在风险, 比如两台实例可能有状态的情况下.
> 
> Maybe this is not the scope of this bugfix.

tcc rpc mode can't find the channel, if no this change.